File: 3-Patron/python/Cliente.py
class Cliente():
    def __init__(self, nombre, apellido, edad, correo, contrasena):
        self.__nombre = nombre
        self.__apellido = apellido
        self.__edad = edad
        self.__correo = correo
        self.__contrasena = contrasena

    @property
    def nombre(self):
        return self.__nombre

    @nombre.setter
    def nombre(self, new_nombre):
        self.__nombre = new_nombre

    @property
    def apellido(self):
        return self.__apellido

    @apellido.setter
    def apellido(self, new_apellido):
        self.__apellido = new_apellido

    @property
    def edad(self):
        return self.__edad

    @edad.setter
    def edad(self, new_edad):
        self.__edad = new_edad

    @property
    def correo(self):
        return self.__correo

    @correo.setter
    def correo(self, new_correo):
        self.__correo = new_correo

    @property
    def contrasena(self):
        return self.__contrasena

    @contrasena.setter
    def contrasena(self, new_contrasena):
        self.__contrasena = new_contrasena

    def autenticar(self):
        pass

File: 4-Patron/php/Cliente.php
<?php
class Cliente
{
  private $nombre;
  private $apellido;
  private $edad;
  private $correo;
  private $contrasena;

  function __construct($nombre,$apellido,$edad,$correo,$contrasena, $skinPersonaje)
  {
    $this->$nombre = $nombre;
    $this->$apellido = $apellido;
    $this->$edad = $edad;
    $this->$correo = $correo;
    $this->$contrasena = $contrasena;
    $this->$skinPersonaje = $skinPersonaje
  }

  function getNombre()
  {
    return $this->nombre;
  }

  function setNombre($nombre)
  {
    $this->nombre = $nombre;
  }

  function getApellido()
  {
    return $this->apellido;
  }

  function setApellido($apellido)
  {
    $this->apellido = $apellido;
  }

  function getEdad()
  {
    return $this->edad;
  }

  function setEdad($edad)
  {
    $this->edad = $edad;
  }

  function getCorreo()
  {
    return $this->correo;
  }

  function setNombre($correo)
  {
    $this->correo = $correo;
  }

  function getContra()
  {
    return $this->contraseña;
  }

  function setContra($contraseña)
  {
    $this->contraseña = $contraseña;
  }

  function autenticar()
  {
    //
  }

  function cambiarSkin($tipo, $ruta)
  {
    $gSkin = new GestorDeSkin();
    $skinPersonaje = $gSkin->cambiarSkin($tipo, $ruta);
  }
}
?>
